cmd/kubeadm/app/apis/kubeadm/bootstraptokenstring.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Note: This file should be kept in sync with the similar one for the external API
+package kubeadm
+
+import (
+	"fmt"
+	"strings"
+
+	bootstrapapi "k8s.io/client-go/tools/bootstrap/token/api"
+	bootstraputil "k8s.io/client-go/tools/bootstrap/token/util"
+)
+
+// BootstrapTokenString is a token of the format abcdef.abcdef0123456789 that is used
+// for both validation of the authenticy of the API server from a joining node's point
+// of view and as an authentication method for the node in the bootstrap phase of
+// "kubeadm join". This token is and should be short-lived
+type BootstrapTokenString struct {
```

Comment:
If I'm not wrong, bootstrap token have their own lifecycle (now GA I think) which is different from kubeadm api lifecycle.
If this is true, we should move `BootstrapTokenString` in `client-go` - out of `apis/kubeadm` - and reference this type both from kubeadm internal API and from kubeadm v1alpha2 API. Does this makes sense to you?

Reply:
It does make sense to me. If we feel comfortable with these types we can do the switch now. Otherwise we'll do it in the next cycle. It has no user-visible effect so we can postpone and take that as the next step I thought (v1beta1 or v1alpha3). Does that make sense to you, to stage this here?

Reply:
Since we both agree on the target, fine for me doing an intermediate step